backend/fastapi_structure.py

```
@@ -0,0 +1,378 @@
+
+# Security setup
+security = HTTPBearer()
+
+# Initialize services - these handle the core business logic
+document_service = DocumentService()
+embedding_service = EmbeddingService()
+rag_service = RAGService()
+ai_agent_service = AIAgentService()
+
+# Authentication dependency - simplified for demo
+async def get_current_user(credentials: HTTPAuthorizationCredentials = Depends(security)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)